presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java

```diff
@@ -158,7 +159,9 @@ public static long arrayPosition(
                 Object arrayValue = type.getObject(array, i);
                 try {
                     Boolean result = (Boolean) equalMethodHandle.invoke(arrayValue, element);
-                    checkNotIndeterminate(result);
+                    if (result == null) {
+                        throw new PrestoException(NOT_SUPPORTED, "array_position does not support element of array type that contains null");
```

Comment:
Maybe something like `array_position does not support elements of complex types that contain null`? In theory `T` can be a map or a row.